packages/connect-react/src/hooks/form-context.tsx

```diff
@@ -173,7 +180,14 @@ export const FormContextProvider = <T extends ConfigurableProps>({
         errs.push("not a string");
       }
     } else if (prop.type === "app") {
-      // TODO need to know about auth type
+      const field = fields[prop.name]
+      if (!field.extra.app) {
+        errs.push("app field not registered")
+      } else if (!value) {
```

Comment:
Only should do this validation if `field.extra.app.authType` is "keys" or "oauth"